client/extensions/woocommerce/state/ui/products/list-reducer.js

```diff
@@ -5,53 +5,43 @@
  */
 
 import { createReducer } from 'state/utils';
+import { get } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import { DEFAULT_QUERY } from 'woocommerce/state/sites/products/utils';
 import {
-	WOOCOMMERCE_PRODUCT_DELETE_SUCCESS,
 	WOOCOMMERCE_PRODUCTS_REQUEST,
 	WOOCOMMERCE_PRODUCTS_REQUEST_SUCCESS,
 } from 'woocommerce/state/action-types';
 
 export default createReducer( null, {
-	[ WOOCOMMERCE_PRODUCT_DELETE_SUCCESS ]: productsDeleteSuccess,
 	[ WOOCOMMERCE_PRODUCTS_REQUEST ]: productsRequest,
 	[ WOOCOMMERCE_PRODUCTS_REQUEST_SUCCESS ]: productsRequestSuccess,
 } );
 
-export function productsRequestSuccess( state, action ) {
-	const prevState = state || {};
-	const { params, products } = action;
-	const page = params.page || null;
-	const productIds = products.map( p => {
-		return p.id;
-	} );
+export function productsRequestSuccess( state = {}, action ) {
+	// If not set in the action, default to the defaults
+	const page = get( action, 'params.page', DEFAULT_QUERY.page );
+	const search = get( action, 'params.search', DEFAULT_QUERY.search );
+
 	return {
-		...prevState,
+		...state,
 		currentPage: page,
-		productIds,
+		currentSearch: search,
 		requestedPage: null,
+		requestedSearch: null,
 	};
 }
 
-export function productsRequest( state, action ) {
-	const prevState = state || {};
-	const { params } = action;
-	const page = params.page || null;
-	return {
-		...prevState,
-		requestedPage: page,
-	};
-}
+export function productsRequest( state = {}, action ) {
+	const page = get( action, 'params.page', null );
```

Comment:
TIL. Here I have been always passing an array as the 2nd parameter when dot notation is so much more compact. :P